### PR TITLE
fixes issue #306 & #302

### PR DESCRIPTION
**Pull request by Tanishashrivas**
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
#302  & #306 

## Type of change

Please give a X on it which is applicable

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**_Test A Describe here_**

**_Test B Describe here (if Requred)_**

# Screenshots and Videos:

Give screenshots and video of the changes you made

# Checklist:
Give a X on, which is applicable

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @Tanishashrivas! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by pooranjoyb** [comment]
Try running `npm run build` in the local to detect the build errors. @Tanishashrivas

**Comment by Tanishashrivas** [comment]
> Try running `npm run build` in the local to detect the build errors. @Tanishashrivas

Done. Please review and merge.

**Comment by Tanishashrivas** [comment]
@pooranjoyb sir? I'm hoping u would merge this today so i can start  working on other issue as well

File: src/pages/Shop/ProductDetail.tsx [hunk 1]
```diff
@@ -138,21 +157,26 @@ function ProductDetail() {
...
                 }
-
+    
                 console.log("Product added to cart:", product);
                 dispatch(addItem({ item: product }));
                 toast.success('Product added to cart');
             }
+    
+            // Navigate the user to the cart
+            navigate('/home/shop/cart');
```
Comment (pooranjoyb) [inline src/pages/Shop/ProductDetail.tsx:172]: Navigating the User to the cart is not needed. Else everything looks good.

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Comment by Tanishashrivas** [comment]
Done!

**Review by pooranjoyb — APPROVED** [review]

**Comment by pooranjoyb** [comment]
Nice work @Tanishashrivas

**Pull request merged**